ext/spl/spl_directory.c

```diff
@@ -196,7 +196,7 @@ static inline void spl_filesystem_object_get_file_name(spl_filesystem_object *in
 		case SPL_FS_INFO:
 		case SPL_FS_FILE:
 			if (!intern->file_name) {
-				php_error_docref(NULL, E_ERROR, "Object not initialized");
```

Comment:
This function seems to be used in a bunch of places. Did you check that they can all handle the exception?